src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java

```diff
@@ -220,32 +414,67 @@ private boolean isPlayerAttacking()
         // proof to new weapons which don't need custom code and the weapon
         // stats are enough.
         Actor target = client.getLocalPlayer().getInteracting();
-        if (target != null)
+        if (target != null && (target instanceof NPC))
         {
             boolean isTargetDummy = target.getHealthRatio() == -1 && target.getCombatLevel() == 0;
             boolean attackingNPC = target.getCombatLevel() > 0 || isTargetDummy;
-            boolean notWalking = animationId != -1;
+            // Not walking is either any player animation or the edge cases which don't trigger an animation, e.g Salamander.
+            boolean notWalking = animationId != -1 || getSalamanderAttack();
```

Comment:
I think this is check for animationId != -1 is causing false positives with the player being flinched. Can we add an exclude list of common animations that are not attacks?

Reply:
Yeah it definitely isn't perfect, but I can say it's already paying off. All the new Emberlight, Bone claws, etc just work. No new code needed for them to be picked up just fine.

Adding more exclusions will only help, flinching/eating and thralls will go along way to ensuring less false positives occur.

Reply:
![java_9ungKII5gx](https://github.com/user-attachments/assets/c64af185-8cfb-4c6a-a184-d14ad27db50c)
![java_MGCzYMoOe9](https://github.com/user-attachments/assets/666212c1-176e-41f5-a941-422a3e538d9a)